extension.ts

```diff
@@ -3,528 +3,27 @@
  * events to their string names and passes them on to ModeHandler via
  * handleKeyEvent().
  */
-import './src/actions/include-all';
-
-import * as vscode from 'vscode';
-import * as path from 'path';
-
-import { CompositionState } from './src/state/compositionState';
-import { EditorIdentity } from './src/editorIdentity';
-import { Globals } from './src/globals';
-import { Jump } from './src/jumps/jump';
-import { ModeHandler } from './src/mode/modeHandler';
-import { ModeHandlerMap } from './src/mode/modeHandlerMap';
-import { Mode } from './src/mode/mode';
-import { Notation } from './src/configuration/notation';
-import { Logger } from './src/util/logger';
-import { Position } from './src/common/motion/position';
-import { StatusBar } from './src/statusBar';
-import { VsCodeContext } from './src/util/vscode-context';
-import { commandLine } from './src/cmd_line/commandLine';
-import { configuration } from './src/configuration/configuration';
-import { globalState } from './src/state/globalState';
-import { taskQueue } from './src/taskQueue';
-import { Register } from './src/register/register';
-
-let extensionContext: vscode.ExtensionContext;
-let previousActiveEditorId: EditorIdentity | undefined = undefined;
-let lastClosedModeHandler: ModeHandler | null = null;
-
-interface ICodeKeybinding {
-  after?: string[];
-  commands?: { command: string; args: any[] }[];
-}
-
-export async function getAndUpdateModeHandler(forceSyncAndUpdate = false): Promise<ModeHandler> {
-  const activeTextEditor = vscode.window.activeTextEditor;
-  const activeEditorId = EditorIdentity.fromEditor(activeTextEditor);
-
-  let [curHandler, isNew] = await ModeHandlerMap.getOrCreate(activeEditorId);
-  if (isNew) {
-    extensionContext.subscriptions.push(curHandler);
-  }
-
-  curHandler.vimState.editor = activeTextEditor!;
-
-  if (
-    forceSyncAndUpdate ||
-    !previousActiveEditorId ||
-    !previousActiveEditorId.isEqual(activeEditorId)
-  ) {
-    curHandler.syncCursors();
-    await curHandler.updateView(curHandler.vimState, { drawSelection: false, revealRange: false });
-  }
-
-  previousActiveEditorId = activeEditorId;
-
-  if (curHandler.vimState.focusChanged) {
-    curHandler.vimState.focusChanged = false;
-
-    if (previousActiveEditorId) {
-      const prevHandler = ModeHandlerMap.get(previousActiveEditorId);
-      prevHandler!.vimState.focusChanged = true;
-    }
-  }
-
-  return curHandler;
-}
+import './src/actions/include-main';
```

Comment:
What's the rationale for splitting this file into `extension.ts` and `extensionBase.ts`?

Reply:
The names are not intuitive and at the beginning I call them `extensionNodeEntry.ts`, `extensionBase.ts` and then we later on we introduce `extensionWebEntry.ts`. The idea is that we will have different entry files for node and web, and the major difference between the entry files are the modules they load and the default flag/options based on the environment. The `extensionBase` will be platform/env agnostic ultimately.

 I didn't make it so now as there are quite a few places (package.json, test, etc) rely on `extension.js` so I try to keep the change minimal.

Reply:
Got it 👍